assets/Product.wxs

```diff
@@ -9,8 +9,10 @@
   <!-- UpgradeCode GUID MUST REMAIN SAME THROUGHOUT ALL VERSIONS, otherwise, updates won't occur. -->
   <?if $(sys.BUILDARCH)=x64?>
     <?define UpgradeCode = "31ab5147-9a97-4452-8443-d9709f0516e1" ?>
+    <?define ProductSemanticVersionWithNameAndOptionalArchitecture = "$(var.ProductName)-$(env.ProductSemanticVersion)"?>
   <?else?>
     <?define UpgradeCode = "1d00683b-0f84-4db8-a64f-2f98ad42fe06" ?>
+    <?define ProductSemanticVersionWithNameAndOptionalArchitecture = "$(var.ProductName)-$(env.ProductSemanticVersion) ($(sys.BUILDARCH))"?>
```

Comment:
Any reason we are using a `-` and a `_` in `ProductVersionWithName`?

If not, could we define `ProductVersionWithName` before this and use it instead of composing this from the components?

Reply:
I don't know `ProductVersionWithName` was like that since 2016 if you do a blame on it. You would need to look at all usages and analyse them if one could change it. But to me it seems unrelated to the bug fix. Maybe we should clean up the installer a bit in a separate PR later?